lib/index.js

```diff
@@ -0,0 +1,122 @@
+const { getInput, info } = require('@actions/core')
+const { context, getOctokit } = require('@actions/github')
+const exec = require('@actions/exec')
+const axios = require('axios').default
+
+// The Checks API limits the number of annotations to a maximum of 50 per API request
+const MAX_ANNOTATIONS_PER_REQUEST = 50
+
+const getSQInfo = (repo) =>  ({
+	projectKey: getInput('projectKey') ? getInput('projectKey') : `${repo.owner}-${repo.repo}`,
+	projectName: getInput('projectName') ? getInput('projectName') : `${repo.owner}-${repo.repo}`,
+	projectBaseDir: getInput('projectBaseDir'),
+	host: getInput('host'),
+	token: getInput('token')
+})
+
+const getScannerCommand = (sonarqube) => 
+	`sonar-scanner -Dsonar.projectKey=${sonarqube.projectKey} -Dsonar.projectName=${sonarqube.projectName} -Dsonar.sources=. -Dsonar.projectBaseDir=${sonarqube.projectBaseDir} -Dsonar.login=${sonarqube.token} -Dsonar.host.url=${sonarqube.host}`
+
+const getSQProjectIssues = async (sonarqube, pageSize, page) => {
+	const tokenb64 = Buffer.from(sonarqube.token + ':').toString('base64')
+
+	try {
+		const response = await axios.get(`${sonarqube.host}/api/issues/search?componentKeys=${sonarqube.projectKey}&statuses=OPEN&ps=${pageSize}&p=${page}`, 
```

Comment:
I would suggest moving the API calls to a separate client file.

Reply:
https://github.com/vtex/tse-cli/blob/master/src/clients/content.ts That might help as a reference